internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -844,47 +827,69 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(log logrus.F
-		mirrorVolumeMounts := []corev1.VolumeMount{
-			{Name: mirrorRegistryCertVolume, MountPath: common.MirrorRegistriesCertificatePath, SubPath: common.MirrorRegistriesCertificateFile},
-			{Name: mirrorRegistryConfVolume, MountPath: common.MirrorRegistriesConfigDir},
-		}
-		mirrorVolumes := []corev1.Volume{
-			{
-				Name: mirrorRegistryCertVolume,
-				VolumeSource: corev1.VolumeSource{
-					ConfigMap: &corev1.ConfigMapVolumeSource{
-						LocalObjectReference: *instance.Spec.MirrorRegistryRef,
-						DefaultMode:          swag.Int32(420),
-						Optional:             swag.Bool(true),
-						Items: []corev1.KeyToPath{
-							{
-								Key:  mirrorRegistryRefCertKey,
-								Path: common.MirrorRegistriesCertificateFile,
+	if mirrorRegConfigMap != nil {
+		// set the registry config
+		if _, ok := mirrorRegConfigMap.Data[mirrorRegistryRefRegistryConfKey]; ok {
+			serviceContainer.VolumeMounts = append(
+				serviceContainer.VolumeMounts,
+				corev1.VolumeMount{
+					Name:      mirrorRegistryConfVolume,
+					MountPath: common.MirrorRegistriesConfigDir,
+				},
+			)
+
+			volumes = append(
+				volumes,
+				corev1.Volume{
+					Name: mirrorRegistryConfVolume,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: *instance.Spec.MirrorRegistryRef,
+							DefaultMode:          swag.Int32(420),
+							Optional:             swag.Bool(true),
+							Items: []corev1.KeyToPath{
+								{
+									Key:  mirrorRegistryRefRegistryConfKey,
+									Path: common.MirrorRegistriesConfigFile,
+								},
 							},
 						},
 					},
 				},
-			},
-			{
-				Name: mirrorRegistryConfVolume,
-				VolumeSource: corev1.VolumeSource{
-					ConfigMap: &corev1.ConfigMapVolumeSource{
-						LocalObjectReference: *instance.Spec.MirrorRegistryRef,
-						DefaultMode:          swag.Int32(420),
-						Optional:             swag.Bool(true),
-						Items: []corev1.KeyToPath{
-							{
-								Key:  mirrorRegistryRefRegistryConfKey,
-								Path: common.MirrorRegistriesConfigFile,
+			)
+		}
+
+		// set the registry cert
+		if _, ok := mirrorRegConfigMap.Data[mirrorRegistryRefCertKey]; ok {
+			serviceContainer.VolumeMounts = append(
+				serviceContainer.VolumeMounts,
+				corev1.VolumeMount{
+					Name:      mirrorRegistryCertVolume,
+					MountPath: common.MirrorRegistriesCertificatePath,
+					SubPath:   common.MirrorRegistriesCertificateFile,
+				},
+			)
+
+			volumes = append(
+				volumes,
+				corev1.Volume{
+					Name: mirrorRegistryCertVolume,
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: *instance.Spec.MirrorRegistryRef,
+							DefaultMode:          swag.Int32(420),
+							Optional:             swag.Bool(true),
+							Items: []corev1.KeyToPath{
+								{
+									Key:  mirrorRegistryRefCertKey,
+									Path: common.MirrorRegistriesCertificateFile,
+								},
```

Comment:
What do you think about changing this to use a single volume (it all comes from the same `ConfigMap` and 2 `VolumeMounts`? 

Was there a specific reason to prefer the 2 volumes that I may be missing?

Reply:
Not particularly. I was hoping to keep as much of the original logic as possible. I think I updated it the most sensible way I could.